src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -0,0 +1,258 @@
+                agentServer.GetReleaseArtifactsFromService(teamProjectId, releaseId).ToList();
+
+            releaseArtifacts.ForEach(x => Trace.Info($"Found Artifact = {x.Alias}"));
+
+            CleanUpArtifactsFolder(executionContext, artifactsWorkingFolder);
+            await DownloadArtifacts(executionContext, releaseArtifacts, artifactsWorkingFolder);
+        }
+
+        private async Task DownloadArtifacts(IExecutionContext executionContext, List<AgentArtifactDefinition> agentArtifactDefinitions, string artifactsWorkingFolder)
+        {
+            Trace.Entering();
+
+            foreach (AgentArtifactDefinition agentArtifactDefinition in agentArtifactDefinitions)
+            {
+                // We don't need to check if its old style artifact anymore. All the build data has been fixed and all the build artifact has Alias now.
+                ArgUtil.NotNullOrEmpty(agentArtifactDefinition.Alias, nameof(agentArtifactDefinition.Alias));
+
+                executionContext.Output(StringUtil.Loc("RMStartArtifactsDownload"));
+                ArtifactDefinition artifactDefinition = ConvertToArtifactDefinition(agentArtifactDefinition, executionContext);
+                executionContext.Output(StringUtil.Loc("RMArtifactDownloadBegin", agentArtifactDefinition.Alias));
+                executionContext.Output(StringUtil.Loc("RMDownloadArtifactType", agentArtifactDefinition.ArtifactType));
+
+                // Get the local path where this artifact should be downloaded. 
+                string downloadFolderPath = GetLocalDownloadFolderPath(agentArtifactDefinition, artifactsWorkingFolder);
+
+                // Create the directory if it does not exist. 
+                if (!Directory.Exists(downloadFolderPath))
+                {
+                    // TODO: old windows agent has a directory cache, verify and implement it if its required.
+                    Directory.CreateDirectory(downloadFolderPath);
+                    executionContext.Output(StringUtil.Loc("RMArtifactFolderCreated", downloadFolderPath));
+                }
+
+                // download the artifact to this path. 
+                RetryExecutor retryExecutor = new RetryExecutor();
+                await retryExecutor.ExecuteAsync(
+                    async () =>
+                        {
+                            //TODO:SetAttributesToNormal
+                            var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
+                            releaseFileSystemManager.CleanupDirectory(downloadFolderPath);
+
+                            if (agentArtifactDefinition.ArtifactType == AgentArtifactType.GitHub
+                                || agentArtifactDefinition.ArtifactType == AgentArtifactType.TFGit
+                                || agentArtifactDefinition.ArtifactType == AgentArtifactType.Tfvc)
+                            {
+                                throw new NotImplementedException();
```

Comment:
This is not implemented. Its coming soon. (old windows agents has this, but not the xplat node agent)